src/main/java/org/opentripplanner/updater/vehicle_parking/VehicleParkingAvailabilityUpdater.java

```diff
@@ -0,0 +1,104 @@
+package org.opentripplanner.updater.vehicle_parking;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.opentripplanner.framework.tostring.ToStringBuilder;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.vehicle_parking.VehicleParking;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingService;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingSpaces;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.service.TransitModel;
+import org.opentripplanner.updater.GraphWriterRunnable;
+import org.opentripplanner.updater.spi.DataSource;
+import org.opentripplanner.updater.spi.PollingGraphUpdater;
+import org.opentripplanner.updater.spi.WriteToGraphCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Graph updater that dynamically sets availability information on vehicle parking lots. This
+ * updater fetches data from a single {@link DataSource<VehicleParking>}.
+ */
+public class VehicleParkingAvailabilityUpdater extends PollingGraphUpdater {
+
+  private static final Logger LOG = LoggerFactory.getLogger(
+    VehicleParkingAvailabilityUpdater.class
+  );
+  private final DataSource<AvailabiltyUpdate> source;
+  private WriteToGraphCallback saveResultOnGraph;
+
+  private final VehicleParkingService vehicleParkingService;
+
+  public VehicleParkingAvailabilityUpdater(
+    VehicleParkingUpdaterParameters parameters,
+    DataSource<AvailabiltyUpdate> source,
+    VehicleParkingService vehicleParkingService
+  ) {
+    super(parameters);
+    this.source = source;
+    this.vehicleParkingService = vehicleParkingService;
+
+    LOG.info("Creating vehicle-parking updater running every {}: {}", pollingPeriod(), source);
+  }
+
+  @Override
+  public void setup(WriteToGraphCallback writeToGraphCallback) {
+    this.saveResultOnGraph = writeToGraphCallback;
+  }
+
+  @Override
+  protected void runPolling() {
+    if (source.update()) {
+      var updates = source.getUpdates();
+
+      var graphWriterRunnable = new UpdateAvailabilities(updates);
+      saveResultOnGraph.execute(graphWriterRunnable);
+    }
+  }
+
+  private class UpdateAvailabilities implements GraphWriterRunnable {
```

Comment:
This class is named like a method.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5979/commits/a2214eb5882b494c0754f097ff7694f619fbf879